src/services/codefixes/helpers.ts

```diff
@@ -1,47 +1,86 @@
         const classMembers = classDeclaration.symbol.members;
         const missingMembers = possiblyMissingSymbols.filter(symbol => !classMembers.has(symbol.getName()));
 
-        let insertion = "";
-
+        let newNodes: Node[] = [];
         for (const symbol of missingMembers) {
-            insertion = insertion.concat(getInsertionForMemberSymbol(symbol, classDeclaration, checker, newlineChar));
+            const newNode = createNewNodeForMemberSymbol(symbol, classDeclaration, checker);
+            if (newNode) {
+                if (Array.isArray(newNode)) {
+                    newNodes = newNodes.concat(newNode);
+                }
+                else {
+                    newNodes.push(newNode);
+                }
+            }
         }
-        return insertion;
+        return newNodes;
     }
 
     /**
      * @returns Empty string iff there we can't figure out a representation for `symbol` in `enclosingDeclaration`.
      */
-    function getInsertionForMemberSymbol(symbol: Symbol, enclosingDeclaration: ClassLikeDeclaration, checker: TypeChecker, newlineChar: string): string {
+    function createNewNodeForMemberSymbol(symbol: Symbol, enclosingDeclaration: ClassLikeDeclaration, checker: TypeChecker): Node[] | Node | undefined {
         const declarations = symbol.getDeclarations();
         if (!(declarations && declarations.length)) {
-            return "";
+            return undefined;
         }
 
         const declaration = declarations[0] as Declaration;
-        const name = declaration.name ? declaration.name.getText() : undefined;
-        const visibility = getVisibilityPrefixWithSpace(getModifierFlags(declaration));
-
-        const type = checker.getTypeOfSymbolAtLocation(symbol, enclosingDeclaration);
+        const name = declaration.name ? getSynthesizedDeepClone(declaration.name) as PropertyName : undefined;
+        const visibilityModifier = createVisibilityModifier(getModifierFlags(declaration));
+        const modifiers = visibilityModifier ? [visibilityModifier] : undefined;
+        const type = checker.getWidenedType(checker.getTypeOfSymbolAtLocation(symbol, enclosingDeclaration));
 
         switch (declaration.kind) {
             case SyntaxKind.GetAccessor:
             case SyntaxKind.SetAccessor:
             case SyntaxKind.PropertySignature:
             case SyntaxKind.PropertyDeclaration:
-                const typeString = checker.typeToString(type, enclosingDeclaration, TypeFormatFlags.None);
-                return `${visibility}${name}: ${typeString};${newlineChar}`;
-
+                const typeNode = checker.createTypeNode(type, enclosingDeclaration);
+                const property = createProperty(
+                    /*decorators*/undefined,
+                    modifiers,
+                    name,
+                    /*questionToken*/ undefined,

```

Comment:
you should be able to know if the parent property is optional or not, and then add the questiontoken as needed.